spring-batch-excel/src/main/java/org/springframework/batch/item/excel/poi/PoiItemReader.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2006-2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.batch.item.excel.poi;
+
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+import org.springframework.batch.item.excel.AbstractExcelItemReader;
+import org.springframework.batch.item.excel.Sheet;
+import org.springframework.core.io.Resource;
+
+/**
+ * {@link org.springframework.batch.item.ItemReader} implementation which uses apache POI to read an Excel
+ * file. It will read the file sheet for sheet and row for row. It is based on
+ * the {@link org.springframework.batch.item.file.FlatFileItemReader}
+ *
+ * @param <T> the type
+ * @author Marten Deinum
+ * @since 0.5.0
+ */
+public class PoiItemReader<T> extends AbstractExcelItemReader<T> {
+
+    private Workbook workbook;
+
+    @Override
+    protected Sheet getSheet(final int sheet) {
+        return new PoiSheet(this.workbook.getSheetAt(sheet));
+    }
+
+    @Override
+    protected int getNumberOfSheets() {
+        return this.workbook.getNumberOfSheets();
+    }
+
+    @Override
+    protected void openExcelFile(final Resource resource) throws Exception {
+        this.workbook = WorkbookFactory.create(resource.getInputStream());
```

Comment:
There, sadly, is no `close()` method on the created workbook. However according to what I've read closing the `InputStream` should fix that. However the earlier solution was faulty, as you commented earlier. For now I keep the used `InputStream` and close it in the `doClose` method.